### PR TITLE
[LTC] Final tweaks on XLATensor and XLAGraphExecutor

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
Fixes some TODOs and adds some comments on XLATensor and XLAGraphExecutor. This is the final move for LTC migration.

Test Plan:
CI.

### CODE REVIEW

File: torch_xla/csrc/xla_graph_executor.cpp [hunk 1]
```diff
@@ -786,7 +785,6 @@ XLAGraphExecutor::ScheduleSyncTensorsGraph(
         TF_VLOG(3) << "Executing IR graph hash "
                    << torch::lazy::HashToString(hash)
                    << " on devices: " << absl::StrJoin(devices, ",");
-        // TODO(jwtan): Remove the WrapXlaData when inherits LazyGraphExecutor.
         results = WrapXlaData(xla::ComputationClient::Get()->ExecuteReplicated(
```
Comment (JackCaoG) [inline torch_xla/csrc/xla_graph_executor.cpp:788]: can we still remove `WrapXlaData`?

**Review by JackCaoG — APPROVED** [review]
Thanks!

**Review by wonjoolee95 — APPROVED** [review]

File: torch_xla/csrc/xla_graph_executor.cpp (hunk 1 above)
Comment (alanwaketan) [reply torch_xla/csrc/xla_graph_executor.cpp:788]: I guess we can once we restructure the xla::ComputationClient to inherit torch::lazy::ComputationClient, but that's later...

**Comment by alanwaketan** [comment]
Thanks Jack and Wonjoo.

**Pull request merged**